cafebabel/static/js/script.js

```diff
@@ -26,6 +26,40 @@ Array.from(document.querySelectorAll('a')).forEach(a => {
     a.setAttribute('target', '_blank')
 })
 
+/* smooth scroll to anchor https://stackoverflow.com/a/17733311/6481285 */
+function animate(elem, style, from, to, time) {
+  if (!elem) {
+    return
+  }
+  var start = new Date().getTime(),
+    timer = setInterval(() => {
+      var step = Math.min(1, (new Date().getTime() - start) / time)
+      elem[style] = from + step * (to - from)
+      if (step === 1) {
+        clearInterval(timer)
+      }
+    }, 25)
+  elem[style] = from
+}
+const subscribeLink = document.querySelector(
+  '#social-networks .icon-newsletter'
+)
+subscribeLink.addEventListener('click', () => {
+  const target = document.querySelector('#static-pages-links')
+  const scrollPosition =
+    window.pageYOffset !== undefined
+      ? window.pageYOffset
+      : (document.documentElement || document.body.parentNode || document.body)
+          .scrollTop
+  animate(
+    document.scrollingElement || document.documentElement,
+    'scrollTop',
+    scrollPosition,
+    target.offsetTop,
+    300
```

Comment:
`/* Total(?) duration of the animation in ms, set to 300 because… */`

Reply:
haha I don't know how to justify 300, well 200 was too fast and 400 to slow I guess